From ad9c13431f955b6b4e7b2e59640b82c021833ae2 Mon Sep 17 00:00:00 2001
From: Peter Chiu <chui-hao.chiu@mediatek.com>
Date: Sat, 28 Sep 2024 10:39:26 +0800
Subject: [PATCH 232/265] mtk: mt76: mt7990: refactor platfrom related
 configuration

1. Use enum to represent txfree path.
1. Use enum to represent bypass rro option.
3. Add some debug logs.

Signed-off-by: Peter Chiu <chui-hao.chiu@mediatek.com>
---
 mt7996/init.c   | 29 ++++++++++++++++++-----------
 mt7996/mt7996.h | 12 ++++++++++++
 2 files changed, 30 insertions(+), 11 deletions(-)

diff --git a/mt7996/init.c b/mt7996/init.c
index 0c675829..362a4fc7 100644
--- a/mt7996/init.c
+++ b/mt7996/init.c
@@ -629,19 +629,19 @@ void mt7996_mac_init(struct mt7996_dev *dev)
 	switch (mt76_chip(&dev->mt76)) {
 	case MT7992_DEVICE_ID:
 		rx_path_type = 7;
-		rro_bypass = 2;
-		txfree_path = 1;
+		rro_bypass = MT7996_RRO_BMC_BYPASS_UC_BASED_ON_WTBL;
+		txfree_path = MT7996_TXFREE_FROM_WA;
 		break;
 	case MT7990_DEVICE_ID:
 		rx_path_type = 7;
-		rro_bypass = 2;
-		txfree_path = 0;
+		rro_bypass = MT7996_RRO_BMC_BYPASS_UC_BASED_ON_WTBL;
+		txfree_path = MT7996_TXFREE_FROM_MAC;
 		break;
 	case MT7996_DEVICE_ID:
 	default:
 		rx_path_type = 2;
-		rro_bypass = 1;
-		txfree_path = 0;
+		rro_bypass = MT7996_RRO_ALL_NOT_BYPASS;
+		txfree_path = MT7996_TXFREE_FROM_MAC;
 		break;
 	}
 
@@ -662,8 +662,8 @@ void mt7996_mac_init(struct mt7996_dev *dev)
 		if (is_mt7996(&dev->mt76)) {
 			/* Bollinger */
 			rx_path_type = 2;
-			rro_bypass = 1;
-			txfree_path = 0;
+			rro_bypass = MT7996_RRO_ALL_NOT_BYPASS;
+			txfree_path = MT7996_TXFREE_FROM_MAC;
 		}
 		break;
 	default:
@@ -672,12 +672,19 @@ void mt7996_mac_init(struct mt7996_dev *dev)
 
 	/* griffin does not have WA */
 	if (!dev->has_rro && mt7996_has_wa(dev))
-		txfree_path = 1;
+		txfree_path = MT7996_TXFREE_FROM_WA;
 
-	mt7996_mcu_set_rro(dev, UNI_RRO_SET_PLATFORM_TYPE, dev->hif2 ? rx_path_type : 0);
-	mt7996_mcu_set_rro(dev, UNI_RRO_SET_BYPASS_MODE, dev->has_rro ? rro_bypass : 3);
+	rx_path_type = dev->hif2 ? rx_path_type : 0;
+	rro_bypass = dev->has_rro ? rro_bypass : MT7996_RRO_ALL_BYPASS;
+
+	mt7996_mcu_set_rro(dev, UNI_RRO_SET_PLATFORM_TYPE, rx_path_type);
+	mt7996_mcu_set_rro(dev, UNI_RRO_SET_BYPASS_MODE, rro_bypass);
 	mt7996_mcu_set_rro(dev, UNI_RRO_SET_TXFREE_PATH, txfree_path);
 
+	dev_info(dev->mt76.dev,
+		"Platform_type = %d, bypass_rro = %d, txfree_path = %d\n",
+		rx_path_type, rro_bypass, txfree_path);
+
 	if (dev->has_rro) {
 		u16 timeout;
 
diff --git a/mt7996/mt7996.h b/mt7996/mt7996.h
index f052a974..722aa5cc 100644
--- a/mt7996/mt7996.h
+++ b/mt7996/mt7996.h
@@ -262,6 +262,18 @@ enum mt7996_rxq_id {
 	MT7996_RXQ_RRO_IND = 0,
 };
 
+enum {
+	MT7996_TXFREE_FROM_MAC,
+	MT7996_TXFREE_FROM_WA,
+};
+
+enum {
+	MT7996_RRO_BMC_NOT_BYPASS_UC_BASED_ON_WTBL,
+	MT7996_RRO_ALL_NOT_BYPASS,
+	MT7996_RRO_BMC_BYPASS_UC_BASED_ON_WTBL,
+	MT7996_RRO_ALL_BYPASS,
+};
+
 struct mt7996_twt_flow {
 	struct list_head list;
 	u64 start_tsf;
-- 
2.45.2

